src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx

```diff
@@ -45,31 +48,35 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
 
     const option = selectedOption[0];
     const optionId = option.id;
+    const defaultDataSource = this.props.uiSettings?.get('defaultDataSource', null) ?? null;
     if (!option.label) {
       try {
         const title = (await getDataSourceById(optionId, this.props.savedObjectsClient)).title;
         if (!title) {
-          this.props.notifications.addWarning(
-            i18n.translate('dataSource.fetchDataSourceError', {
-              defaultMessage: `Data source with id ${optionId} is not available`,
-            })
-          );
+          this.handleDataSourceFetchError(optionId);
         } else {
           if (!this._isMounted) return;
           this.setState({
             selectedOption: [{ id: optionId, label: title }],
+            defaultDataSource,
           });
         }
       } catch (error) {
-        this.props.notifications.addWarning(
-          i18n.translate('dataSource.fetchDataSourceError', {
-            defaultMessage: `Failed to fetch data source due to ${error}`,
-          })
-        );
+        this.handleDataSourceFetchError(optionId, error);
       }
     }
   }
 
+  handleDataSourceFetchError(optionId: string, error?) {
+    this.props.notifications.addWarning(
+      i18n.translate('dataSource.fetchDataSourceError', {
+        defaultMessage: error
+          ? `Failed to fetch data source due to ${error}`
```

Comment:
Noticed that the error messages we display is getting diverged among components, can we extract datasourceFetchError as util function to be used by all components?

Reply:
okay